ofono2mm/mm_modem_simple.py

```diff
@@ -0,0 +1,30 @@
+from dbus_next.service import (ServiceInterface,
+                               method, dbus_property, signal)
+from dbus_next.constants import PropertyAccess
+from dbus_next import Variant
+
+class MMModemSimpleInterface(ServiceInterface):
+    def __init__(self, mm_modem):
+        super().__init__('org.freedesktop.ModemManager1.Modem.Simple')
+        self.mm_modem = mm_modem
+
+    @method()
+    async def Connect(self, properties: 'a{sv}') -> 'o':
+        for b in self.mm_modem.bearers:
+            if self.mm_modem.bearers[b].props['Properties'].value['apn'] == properties['apn']:
+                await self.mm_modem.bearers[b].add_auth_ofono(properties['username'] if 'username' in properties else '', 
```

Comment:
Variants should be unwrapped here, otherwise selecting a custom username won't work